pkg/agent/externalnode/external_node_controller_linux_test.go

```diff
@@ -0,0 +1,1155 @@
+
+	err := c.moveIFConfigurations(&config.AdapterNetConfig{MTU: 1500}, "if1", "if2")
+	assert.Equal(t, linkSetMTUError, err)
+
+	defer mockLinkSetMTU(nil)()
+	defer mockLinkSetUp(linkSetUpError)()
+	err = c.moveIFConfigurations(&config.AdapterNetConfig{MTU: 1500}, "if1", "if2")
+	assert.Equal(t, linkSetUpError, err)
+
+	defer mockLinkSetUp(nil)()
+	defer mockRemoveLinkIPs(removeLinkIPsError)()
+	err = c.moveIFConfigurations(&config.AdapterNetConfig{MTU: 1500}, "if1", "if2")
+	assert.Equal(t, removeLinkIPsError, err)
+
+	defer mockRemoveLinkIPs(nil)()
+	defer mockRemoveLinkRoutes(removeLinkRoutesError)()
+	err = c.moveIFConfigurations(&config.AdapterNetConfig{MTU: 1500}, "if1", "if2")
+	assert.Equal(t, removeLinkRoutesError, err)
+}
+
+func mockHostInterfaceExists(exists bool) func() {
+	originalHostInterfaceExists := hostInterfaceExists
+	hostInterfaceExists = func(ifName string) bool {
+		return exists
+	}
+	return func() {
+		hostInterfaceExists = originalHostInterfaceExists
+	}
+}
+
+func mockConfigureLinkRoutes(configureLinkRoutesErr error) func() {
+	originalConfigureLinkRoutes := configureLinkRoutes
+	configureLinkRoutes = func(link netlink.Link, routes []interface{}) error {
+		return configureLinkRoutesErr
+	}
+	return func() {
+		configureLinkRoutes = originalConfigureLinkRoutes
+	}
+}
+
+func mockConfigureLinkAddresses(configureLinkAddressesErr error) func() {
+	originalRemoveLinkRoutes := configureLinkAddresses
+	configureLinkAddresses = func(idx int, ipNets []*net.IPNet) error {
+		return configureLinkAddressesErr
+	}
+	return func() {
+		configureLinkAddresses = originalRemoveLinkRoutes
+	}
+}
+
+func mockRemoveLinkRoutes(removeLinkRoutesErr error) func() {
+	originalRemoveLinkRoutes := removeLinkRoutes
+	removeLinkRoutes = func(link netlink.Link) error {
+		return removeLinkRoutesErr
+	}
+	return func() {
+		removeLinkRoutes = originalRemoveLinkRoutes
+	}
+}
+
+func mockRemoveLinkIPs(removeLinkIPsErr error) func() {
+	originalRemoveLinkIPs := removeLinkIPs
+	removeLinkIPs = func(link netlink.Link) error {
+		return removeLinkIPsErr
+	}
+	return func() {
+		removeLinkIPs = originalRemoveLinkIPs
+	}
+}
+
+func mockLinkSetUp(linkSetUpErr error) func() {
+	originalLinkSetUp := linkSetUp
+	linkSetUp = func(link netlink.Link) error {
+		return linkSetUpErr
+	}
+	return func() {
+		linkSetUp = originalLinkSetUp
+	}
+}
+
+func mockLinkSetMTU(linkSetMTUErr error) func() {
+	originalLinkSetMTU := linkSetMTU
+	linkSetMTU = func(link netlink.Link, mtu int) error {
+		return linkSetMTUErr
+	}
+	return func() {
+		linkSetMTU = originalLinkSetMTU
+	}
+}
+
+func mockLinkByName(t *testing.T, calledTimes int) func() {
+	originalLinkByName := linkByName
+	counter := 0
+	linkByName = func(_ string) (netlink.Link, error) {
+		counter++
+		return &netlink.Dummy{}, nil
+	}
+	return func() {
+		if counter != calledTimes {
+			t.Errorf("mockLinkByName should be called %d times, but was actually called %d times", calledTimes, counter)
```

Comment:
We should first reset the function, then check with the times of the expected call

Reply:
updated